DDCore/src/DetectorImp.cpp

```diff
@@ -642,15 +642,20 @@ Handle<NamedObject> DetectorImp::getRefChild(const HandleMap& e, const string& n
       const void* other;
       ptr(const void* p) { other = p; }
     };
-    int cnt = 0;
     std::string nam = "";
     ptr m(&e), ref(this);
     if ( ref.c > m.c && m.c < ref.c+sizeof(*this) ) nam = m.omap->name;
-    cout << "GetRefChild: Failed to find child with name: '" << name
-	 << "'. Map " << nam << " contains " << e.size() << " elements." << endl;
-    for(i=e.begin(); i!=e.end(); ++i)
-      cout << "   " << cnt << "  " << (*i).first << endl;
-    throw runtime_error(nam+": Cannot find a child with the reference name:" + name);
+    std::stringstream err;
+    err << "getRefChild: Failed to find child with name: " << name
+        << " Map " << nam << " contains " << e.size() << " elements: {";
+    for (it = e.begin(); it != e.end(); ++it) {
+      if (it != e.begin()) {
+        err << ", " << endl;
+      }
+      err << it->first << endl;
```

Comment:
Did you want an `endl` here? This results in e.g. 
```
[2023-05-06 07:41:33.824] [JEventProcessorPODIO] [error] Omitting PODIO collection 'EcalBarrelScFiRecHits' due to exception: getRefChild: Failed to find child with name: EcalBarrelScFiHits Map contains 28 elements: {B0ECalHits
, 
B0TrackerHits
, 
DIRCBarHits
, 
DRICHHits
, 
EcalBarrelSciGlassHits
, 
EcalEndcapNHits
, 
EcalEndcapPHits
, 
EcalEndcapPInsertHits
, 
ForwardOffMTrackerHits
, 
ForwardRomanPotHits
, 
HcalBarrelHits
, 
HcalEndcapNHits
, 
HcalEndcapPInsertHits
, 
LFHCALHits
,
...
```

Reply:
Good catch, fixed.